pkg/plugin/plugin.go

```diff
@@ -0,0 +1,253 @@
+
+	"github.com/lima-vm/lima/v2/pkg/usrlocalsharelima"
+)
+
+type Plugin struct {
+	Name        string `json:"name"`
+	Path        string `json:"path"`
+	Description string `json:"description,omitempty"`
+}
+
+func DiscoverPlugins() ([]Plugin, error) {
+	var plugins []Plugin
+	seen := make(map[string]bool)
+
+	dirs := getPluginDirectories()
+
+	for _, dir := range dirs {
+		pluginsInDir := scanDirectory(dir)
+		for _, plugin := range pluginsInDir {
+			if !seen[plugin.Name] {
+				plugins = append(plugins, plugin)
+				seen[plugin.Name] = true
+			}
+		}
+	}
+
+	slices.SortFunc(plugins,
+		func(i, j Plugin) int {
+			return cmp.Compare(i.Name, j.Name)
+		})
+
+	return plugins, nil
+}
+
+func getPluginDirectories() []string {
+	var dirs []string
+
+	selfDirs := usrlocalsharelima.SelfDirs()
+	dirs = append(dirs, selfDirs...)
+
+	pathEnv := os.Getenv("PATH")
+	if pathEnv != "" {
+		pathDirs := filepath.SplitList(pathEnv)
+		dirs = append(dirs, pathDirs...)
+	}
+
+	if libexecDir, err := usrlocalsharelima.LibexecLima(); err == nil {
+		if _, err := os.Stat(libexecDir); err == nil {
+			dirs = append(dirs, libexecDir)
+		}
+	}
+
+	return dirs
+}
+
+// isWindowsExecutableExt checks if the given extension is a valid Windows executable extension
+// according to PATHEXT environment variable.
+func isWindowsExecutableExt(ext string) bool {
+	if runtime.GOOS != "windows" {
+		return false
+	}
+
+	pathExt := os.Getenv("PATHEXT")
+	if pathExt == "" {
+		pathExt = ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL"
+	}
+
+	extensions := strings.Split(strings.ToUpper(pathExt), ";")
+	extUpper := strings.ToUpper(ext)
+
+	for _, validExt := range extensions {
+		if validExt == extUpper {
+			return true
+		}
+	}
+	return false
+}
+
+func scanDirectory(dir string) []Plugin {
+	var plugins []Plugin
+
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Debugf("Plugin discovery: failed to scan directory %s: %v", dir, err)
+
+		return plugins
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		name := entry.Name()
+		if !strings.HasPrefix(name, "limactl-") {
+			continue
+		}
+
+		pluginName := strings.TrimPrefix(name, "limactl-")
+
+		if strings.Contains(pluginName, ".") {
+			ext := filepath.Ext(name)
+			if !isWindowsExecutableExt(ext) {
+				continue
+			}
+
+			pluginName = strings.TrimSuffix(pluginName, ext)
+		}
+
+		fullPath := filepath.Join(dir, name)
+
+		if !isExecutable(fullPath) {
+			continue
+		}
+
+		plugin := Plugin{
+			Name: pluginName,
+			Path: fullPath,
+		}
+
+		if desc := extractDescFromScript(fullPath); desc != "" {
+			plugin.Description = desc
+		}
+
+		plugins = append(plugins, plugin)
+	}
+
+	return plugins
+}
+
+func isExecutable(path string) bool {
+	info, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+
+	mode := info.Mode()
+	if mode&0o111 != 0 {
+		return true
+	}
+
+	if runtime.GOOS == "windows" {
+		ext := filepath.Ext(path)
+		return isWindowsExecutableExt(ext)
+	}
```

Comment:
I'm not sure if this code is needed.

The file extension is only needed when you run a command without an extension. E.g. when you run `tar` Windows will try each extension from `PATHEXT` in turn to find an executable: `tar.exe`, `tar.bat`, etc.

But Windows will execute any command that exists as-is, regardless of the extension:

```console
C:\Users\suse>tar
tar: Must specify one of -c, -r, -t, -u, -x

C:\Users\suse>copy C:\Windows\System32\tar.exe hello.world
        1 file(s) copied.

C:\Users\suse>hello.world
hello.world: Must specify one of -c, -r, -t, -u, -x
```

So I think I would just remove this block completely. We want to strip the common extensions from the plugin name for the `--help` output, but here we don't really care.